docs/rules.md

```diff
@@ -35,6 +35,55 @@ extended configuration file as well, to pass them both to the TypeScript compile
+</pre>
+
+Implementation rule behind the ts_project macro.
+    Most users should use [ts_project](#ts_project) instead.
+    
+
+**ATTRIBUTES**
+
+
+| Name  | Description | Type | Mandatory | Default |
+| :------------- | :------------- | :------------- | :------------- | :------------- |
+| <a id="ts_project_rule-name"></a>name |  A unique name for this target.   | <a href="https://bazel.build/docs/build-ref.html#name">Name</a> | required |  |
+| <a id="ts_project_rule-allow_js"></a>allow_js |  https://www.typescriptlang.org/tsconfig#allowJs   | Boolean | optional | False |
+| <a id="ts_project_rule-args"></a>args |  https://www.typescriptlang.org/docs/handbook/compiler-options.html   | List of strings | optional | [] |
+| <a id="ts_project_rule-buildinfo_out"></a>buildinfo_out |  Location in bazel-out where tsc will write a <code>.tsbuildinfo</code> file   | <a href="https://bazel.build/docs/build-ref.html#labels">Label</a> | optional |  |
+| <a id="ts_project_rule-composite"></a>composite |  https://www.typescriptlang.org/tsconfig#composite   | Boolean | optional | False |
+| <a id="ts_project_rule-data"></a>data |  Runtime dependencies to include in binaries/tests that depend on this TS code   | <a href="https://bazel.build/docs/build-ref.html#labels">List of labels</a> | optional | [] |
+| <a id="ts_project_rule-declaration"></a>declaration |  https://www.typescriptlang.org/tsconfig#declaration   | Boolean | optional | False |
+| <a id="ts_project_rule-declaration_dir"></a>declaration_dir |  Always controlled by Bazel. https://www.typescriptlang.org/tsconfig#declarationDir   | String | optional | "" |
+| <a id="ts_project_rule-declaration_map"></a>declaration_map |  https://www.typescriptlang.org/tsconfig#declarationMap   | Boolean | optional | False |
+| <a id="ts_project_rule-deps"></a>deps |  Other targets which produce TypeScript typings   | <a href="https://bazel.build/docs/build-ref.html#labels">List of labels</a> | optional | [] |
+| <a id="ts_project_rule-emit_declaration_only"></a>emit_declaration_only |  https://www.typescriptlang.org/tsconfig#emitDeclarationOnly   | Boolean | optional | False |
+| <a id="ts_project_rule-extends"></a>extends |  https://www.typescriptlang.org/tsconfig#extends   | <a href="https://bazel.build/docs/build-ref.html#labels">Label</a> | optional | None |
+| <a id="ts_project_rule-incremental"></a>incremental |  https://www.typescriptlang.org/tsconfig#incremental   | Boolean | optional | False |
+| <a id="ts_project_rule-js_outs"></a>js_outs |  Locations in bazel-out where tsc will write <code>.js</code> files   | List of labels | optional |  |
+| <a id="ts_project_rule-map_outs"></a>map_outs |  Locations in bazel-out where tsc will write <code>.js.map</code> files   | List of labels | optional |  |
+| <a id="ts_project_rule-out_dir"></a>out_dir |  Always controlled by Bazel. https://www.typescriptlang.org/tsconfig#outDir   | String | optional | "" |
```

Comment:
"Always controlled by Bazel" but we still want to document it?

Reply:
I also don't see where this string came from?

Reply:
I'll just remove that part about controlled by Bazel

Reply:
As long as users won't think they should/can use it 🤷